packages/labs/react/src/create-component.ts

```diff
@@ -25,27 +25,24 @@ const listenedEvents: WeakMap<
  */
 const addOrUpdateEventListener = (
   node: Element,
-  event: string,
-  listener: (event?: Event) => void
+  eventName: string,
+  listener?: EventListener
 ) => {
   let events = listenedEvents.get(node);
   if (events === undefined) {
     listenedEvents.set(node, (events = new Map()));
   }
-  let handler = events.get(event);
+
+  const prevListener = events.get(eventName);
+  if (prevListener !== undefined) {
+    node.removeEventListener(eventName, prevListener);
+  }
+
   if (listener !== undefined) {
-    // If necessary, add listener and track handler
-    if (handler === undefined) {
-      events.set(event, (handler = {handleEvent: listener}));
-      node.addEventListener(event, handler);
-      // Otherwise just update the listener with new value
-    } else {
-      handler.handleEvent = listener;
-    }
-    // Remove listener if one exists and value is undefined
-  } else if (handler !== undefined) {
-    events.delete(event);
-    node.removeEventListener(event, handler);
+    events.set(eventName, listener);
+    node.addEventListener(eventName, listener);
```

Comment:
This will thrash `removeEventListener`/`addEventListener` in cases where the event handler is an inline closure, like:

```tsx
<MyElement onFoo={(e) => console.log(e)}></MyElement>
```

This has a measurable perf overhead in some browsers and is why we only call `removeEventListener()` when the listener goes from defined to undefined and `addEventListener()` when the listener goes from undefined to defined.

Reply:
I see! That makes sense.

I thrashing is the default behavior of react anyways no? LIke `add/removeEventListnener` with an inline function would thrash every render.

So we're opting out of react behavior to provide a more performant option?

Reply:
So we're using an `EventListenerObject` to provide a context to attach and swap an `EventListener` without relying on the `removeEventListener` when handlers change but are undefined.

Reply:
AFAIU, React uses event delegation and won't thrash add/removeEventListnener. Since we're handling events outside of React we need to take precautions ourselves. We do the same thing in lit-html's `@event` binding support.

Reply:
You're correct! React does the event hub thing and it'd be cheaper than add/remove :) my bad was more clarifying for my understanding. also cool! did not know that about event binding